### ISSUE TITLE
Implement types

### ISSUE THREAD

**Issue by alanshaw** [tableflip/ipfs-postmsg-proxy#20]
https://github.com/ipfs/interface-ipfs-core/blob/master/SPEC/TYPES.md

**Comment by olizilla** [comment]
Yes please! Without them, we can't rely on it being available when we use `ipfs-redux-bundle` or the TBD `ipfs-provider` lib. See https://github.com/ipfs-shipyard/ipfs-share-files/pull/44/files#r235666061 for a use-case
